build-system/tasks/performance/measure-documents.js

```diff
@@ -0,0 +1,209 @@
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const {
+  CONTROL,
+  EXPERIMENT,
+  RESULTS_PATH,
+  urlToCachePath,
+} = require('./helpers');
+
+// Require Puppeteer dynamically to prevent throwing error in Travis
+let puppeteer;
+
+function requirePuppeteer_() {
+  puppeteer = require('puppeteer');
+}
+
+/**
+ * Setup measurement on page before navigating to the URL. Performance
+ * observers need to be initialized before content begins to load to take
+ * measurements.
+ *
+ * @param {Puppeteer.page} page
+ * @return {Promise} Resolves when script is evaluated
+ */
+const setupMeasurement = page =>
+  page.evaluateOnNewDocument(() => {
+    window.longTasks = [];
+    window.cumulativeLayoutShift = 0;
+    window.measureStarted = Date.now();
+    window.largestContentfulPaint = 0;
+
+    const longTaskObserver = new PerformanceObserver(list =>
+      list.getEntries().forEach(entry => window.longTasks.push(entry))
+    );
+
+    longTaskObserver.observe({entryTypes: ['longtask']});
+
+    const layoutShiftObserver = new PerformanceObserver(list =>
+      list
+        .getEntries()
+        .forEach(entry => (window.cumulativeLayoutShift += entry.value))
+    );
+
+    layoutShiftObserver.observe({entryTypes: ['layout-shift']});
+
+    const largestContentfulPaintObserver = new PerformanceObserver(list => {
+      const entries = list.getEntries();
+      const entry = entries[entries.length - 1];
+      window.largestContentfulPaint = entry.renderTime || entry.loadTime;
+    });
+
+    largestContentfulPaintObserver.observe({
+      entryTypes: ['largest-contentful-paint'],
+    });
+  });
+
+/**
+ * Evaluate script on the page to collect and calculate metrics
+ *
+ * @param {Puppeteer.page} page
+ * @return {Promise<object>} Resolves with page load metrics
+ */
+const readMetrics = page =>
+  page.evaluate(() => {
+    const entries = performance.getEntries();
+
+    function getMetric(name) {
+      const entry = entries.find(entry => entry.name === name);
+      return entry ? entry.startTime : 0;
+    }
+
+    const firstPaint = getMetric('first-paint');
+    const firstContentfulPaint = getMetric('first-contentful-paint');
+
+    function getMaxFirstInputDelay() {
+      let longest = 0;
+
+      window.longTasks.forEach(longTask => {
+        if (
+          longTask.startTime > firstContentfulPaint &&
+          longTask.duration > longest
+        ) {
+          longest = longTask.duration;
+        }
+      });
+
+      return longest;
+    }
+
+    function getTimeToInteractive() {
+      return Date.now() - window.measureStarted;
+    }
+
+    return {
+      visible: getMetric('visible'),
+      firstPaint,
+      firstContentfulPaint,
+      largestContentfulPaint: window.largestContentfulPaint,
+      timeToInteractive: getTimeToInteractive(),
+      maxFirstInputDelay: getMaxFirstInputDelay(),
+      cumulativeLayoutShift: window.cumulativeLayoutShift,
```

Comment:
I believe this is a percentage with value from `0` to `1` so let's multiply it by `100` otherwise it'll always get rounded up to `0` in the printed table

Reply:
"The layout shift (LS) score is equal to the impact fraction multiplied by the distance fraction" per entry, and we sum all the entries together here. I think you are correct that typically this is going to be a low number, although not necessarily between 0-1. Will multiply by 100 for now although not 100% sure that's the easiest solution